src/pages/patientView/mutation/PatientViewMutationTable.tsx

```diff
@@ -427,15 +449,30 @@ export default class PatientViewMutationTable extends React.Component<PatientVie
     }
 
     @computed private get columns():Column<Mutation[]>[] {
-        return this.props.columns.map((columnEnum:MutationTableColumn)=>this._columns[columnEnum]);
+        return this.props.columns.reduce((columns:Column<Mutation[]>[], next:MutationTableColumn)=>{
+            let shouldAdd = true;
+            if (next === MutationTableColumn.MRNA_EXPR &&
+                (!this.props.mrnaExprRankGeneticProfileId
+                || this.getSamples().length > 1)) {
+                shouldAdd = false;
+            } else if (next === MutationTableColumn.TUMORS && this.getSamples().length < 2) {
+                shouldAdd = false;
+            } else if (next === MutationTableColumn.COPY_NUM && !this.props.discreteCNAGeneticProfileId) {
+                shouldAdd = false;
+            }
+            if (shouldAdd) {
```

Comment:
How about extending the default `Column` definition with and optional `shouldAdd` function and handle each column in a separate function? (If no `shouldAdd` provided, it should always add that column)

```JavaScript
MutationTableColumn<T> extends Column<T>
{
    shouldAdd: (...) => boolean;
}
...
...
if (column.shouldAdd(this.props.store)) {
    columns.push(...);
}
```